### DISCUSSION TITLE
How to store user credentials

### DISCUSSION THREAD

**Discussion by helghast79** [tauri-apps/tauri#1222] [Q&A]
Is there a way to store user's credentials or user sensitive data that should be persisted like login credentials? Local storage only saves data while the app is running and is lost once the app is closed. [Keytar](http://atom.github.io/node-keytar/) works in electron, is there a way to implemented it in Tauri?

**Comment by Kelerchian** [comment]
Keytar has a rust binding https://docs.rs/keytar/0.1.3/keytar/

I think what you can do now is to manually pass the credential to the rust-runtime and then use keytar in the rust side.

**Comment by nothingismagick** [comment]
We will be releasing an official plugin for Tauri that integrates with [Stronghold](https://github.com/iotaledger/stronghold.rs).

**Reply by nothingismagick** [reply]
Here are the new plugins:

Using a yubikey:
https://github.com/tauri-apps/tauri-plugin-authenticator

Actually useful for secret storage:
https://github.com/tauri-apps/tauri-plugin-stronghold
